### ISSUE TITLE
move music_assistant.db out of /config (but remain on the HA instance MA is running on)

### ISSUE THREAD

**Issue by Mariusthvdb** [music-assistant/hass-music-assistant#341]
Please consider support for allowing the music_assistant.db to be saved elsewhere, to keep the `/config` share as clean as possible.

In Discord we briefly discussed the technique Powercalc supports doing this, by allowing a user configured folder inside the CC folder. Used so this is not wiped at updating the CC.

For my personal use it would be sufficient to make that a fixed `/db` folder, so `/config/custom_components/mass/db`

I noticed others mention a location even outside the HA instance, because the file might get large quite soon. I would no be an advocate for that, as it would make MA rely on an external source per definition, and I dont think that would be very robust.

thanks for considering!

**Comment by erkr** [comment]
I really would like to have this improvement in place. Now the database (200mb) is part of my nightly backups as they include the config folder. Moving the database to i.e. my media folder would solve that.

**Comment by marcelveldt** [comment]
Moving the db to a location outside of HA (e.g. mounted disk) is very dangerous. I'd rather invest in mysql compatibility in that case. That said. What is 200mb nowadays ? Worst case scenario (when you have like 500.000 items in your library) the db would grow to like 500mb ....

Can we merge the 2 feature requests into one ?
Make it more generic to customize the db path ?
It truly is just a sqlalchemy string so it supports both sqlite in a folder of choice and mysql or postgres

**Comment by erkr** [comment]
> Moving the db to a location outside of HA (e.g. mounted disk) is very dangerous

Agreed to keep it local (my media folder is not a mount, but part of my local SSD). 
But the proposal for a fixed '/db' folder, like '/config/custom_components/mass/db' will keep it part of my backups. The database doesn't need to be backed up (easily regenerated). It only increases my backup times and cloud space.

Note: the Google drive backup add-on, that I use, only allows to exclude particular add-ons and folders like media and shared. 
That is why I was thinking to use the media folder (local, nit part of my backup). No other reason

**Comment by GoSpursGoNL** [comment]
I currently use owntone (formerly called forked-daapd) for library streaming and there I configured the database to be in /share which isn't backed up. Would be great indeed if music assistant would also support this.

**Comment by erkr** [comment]
Moved to discussions: raise your votes 😀
